### ISSUE TITLE
substitutions in dl_distance time complexity

### ISSUE THREAD

**Issue by raphael-sch** [nadvornix/python-fizzle#1]
If you run dl_distance in a time critical loop and use substitutions, you don't want to import defaultdict and iterate through all subs every time. Instead dl_distance should just take a dict as substitution parameter and let 'key in dict' decide to return value or 1.

**Comment by nadvornix** [comment]
What do you think about [this](https://github.com/nadvornix/python-fizzle/commit/ee6ec758e1ff499405a746864003cc9b9c2b1dc0) way of fixing it?

**Comment by raphael-sch** [comment]
yes in general this is how I would do it. In #5 i suggest a few changes to your commit.
